tests/Api/ResourceClassResolverTest.php

```diff
@@ -38,6 +39,18 @@ public function testGetResourceClassWithIntendedClassName()
         $this->assertEquals($resourceClass, Dummy::class);
     }
 
+    public function testGetResourceClassWithOtherClassName()
+    {
+        $dummy = new Dummy();
+        $dummy->setName('Smail');
+        $resourceNameCollectionFactoryProphecy = $this->prophesize(ResourceNameCollectionFactoryInterface::class);
+        $resourceNameCollectionFactoryProphecy->create()->willReturn(new ResourceNameCollection([Dummy::class]))->shouldBeCalled();
+
+        $resourceClassResolver = new ResourceClassResolver($resourceNameCollectionFactoryProphecy->reveal());
+        $resourceClass = $resourceClassResolver->getResourceClass($dummy, DummyCar::class, true);
+        $this->assertEquals($resourceClass, Dummy::class);
+    }
```

Comment:
For example I don't see how `$resourceClassResolver->getResourceClass($dummy, DummyCar::class, true)` would be called as `DummyCar` isn't a part of your `ResourceNameCollection` and isn't the `Dummy` object. I may miss something.

Reply:
The actual behavior that I have is that I have a resource `Book` linked to a  `File` entity which is not a resource but the file entity is linked to an `Author` which is a resource. When I call the route I got an exceptio nfrom the ResourceClassresolver saying that Author is not a resource.
The JsonLd ItemNormalizer call get resourceClass [here](https://github.com/api-platform/core/blob/master/src/JsonLd/Serializer/ItemNormalizer.php#L65), the first param is my Author object and the second one would be the Book.

I was hesitant to send the PR as it does fix my issue, doesn't break any existing tests but I understand that it might come from somewhere else....

Reply:
So (`R` represents a resource):

```
Book (R) => File
File => Author (R)
```

You request `Book`, then it normalizes the `File` followed by the `Author`? So call would be `getResourceClass(Author::class (the class being denormalized), Book::class (the context resource class is correct), true);`.

Am I correct?

I'm not sure it comes from elsewhere so fixing this here would be correct. I just want to be sure I get this correctly to give this a try.

Reply:
True that what is called